### ISSUE TITLE
Truncated/corrupted jpeg2000 image

### ISSUE THREAD

**Issue by Pazaak** [python-pillow/Pillow#3946]
Hi, I have a batch of images like this and I want them converted to a numpy array:
https://drive.google.com/open?id=1hx_s1LZV48ID89CFaTzE3vjUTXN1IcGl

The images seem to be jpeg2000 and truncated. Trying to load them without LOAD_TRUNCATED_IMAGES gives me the usual exception:
```python
from PIL import Image
a = Image.open('broken.jpg').convert("RGB")
```
OSError: broken data stream when reading image file

Using `LOAD_TRUNCATED_IMAGES = True`
```python
from PIL import Image, ImageFile
import numpy as np
ImageFile.LOAD_TRUNCATED_IMAGES = True
a = np.asarray(Image.open('broken.jpg').convert("RGB"))
print(np.sum(a))
```
Gives a zero sum of the array, indicating that the image is not properly loaded. I was expecting an array containing the RGB information of the image.

* OS: Ubuntu 16.04
* Python: 3.68
* Pillow: 6.1.0

Thank you!

**Comment by radarhere** [comment]
It is registering as truncated because the JPEG2000 decoder is returning an error - different from the more common truncated image errors that are raised directly from ImageFile.

https://github.com/python-pillow/Pillow/blob/575a616bb594b974cdc71a87b128b522560a3339/src/libImaging/Jpeg2KDecode.c#L622-L626

**Comment by radarhere** [comment]
Testing, I find this is resolved by #4996
